text/0821-public-types.md

```diff
@@ -392,6 +393,28 @@ function takesRouteInfoWithAttributes(routeInfoWithAttributes) {
 ```
 
 
+### `Resolver`
+
+The resolver is a contract implemented by libraries outside Ember itself, such as `ember-resolver`, `ember-strict-resolver`, and any number of custom resolvers which exist in apps across the ecosystem. It has never had public documentation, but is fully public API. It is a user-constructible interface with the following definition (using the `Factory` and `FullName` types exported from the new `@ember/owner` module), available as a new named type-only export from the `@ember/engine` package:
+
+```ts
+import EmberObject from '@ember/object';
+import type { Factory, FullName } from '@ember/owner';
+
+type KnownForTypeResult<Name extends string> = {
+  [fullName in `${Name}:${string}`]: boolean | undefined;
+};
+
+export interface Resolver extends EmberObject {
```

Comment:
Does a `Resolver` implementation have to be an `EmberObject`, or do the existing implementations just happen to be?

Reply:
Mmm, good catch. I don’t *think* it has to be, and I made the mistake of pulling this over from the existing `ember-resolver` types—that one is, of course, but I don’t think the contract does or should require it. Will update!